shimmingtoolbox/dicom_to_nifti.py

```diff
@@ -0,0 +1,128 @@
+import os
+import numpy as np
+from .read_nii import read_nii
+
+
+def dicom_to_nifti(unsortedDicomDir, niftiPath):
+
+    #os.mkdir(niftiPath)
+    print(unsortedDicomDir)
+    print(niftiPath)
+
+    # Check for which "which" to use depending on OS
+    if os.name == 'nt':  # If OS is Windows
+        which = 'where'
+        copy = 'copy'
+    else:
+        which = 'which'
+        copy = 'cp -r'
+
+    # Make sur dcm2niix is installed
+    if os.system(which + ' dcm2niix') != 0:
+        print('Error: dcm2niix is not installed.')
+
+    # Make sure dcm2bids is installed
+    if os.system(which + ' dcm2bids') != 0:
+        raise Exception('Error: dcm2bids is not installed.')
+
+    # Create bids structure for data
+    participant = ''
+    if os.system('dcm2bids_scaffold -o ' + niftiPath) != 0:
+        raise Exception('Error: dcm2bids_scaffold')
+
+    # Add original data to niftiPath/sourcedata
+    if os.system(copy + ' ' + unsortedDicomDir + ' ' + os.path.join(niftiPath, 'sourcedata')) != 0:
+        raise Exception('Error: copy')
+
+    # Call the dcm2bids_helper
+    if os.system('dcm2bids_helper -d ' + unsortedDicomDir + ' -o ' + niftiPath) != 0:
+        raise Exception('Error: dcm2bids_helper')
+
+    # Check if there is data
+    helperPath = os.path.join(niftiPath, 'tmp_dcm2bids', 'helper')
+    if not os.path.isdir(helperPath):
+        raise Exception('Error: dcm2bids_helper could not create directory helper')
+
+    # Make sure there is data in niftiPath / tmp_dcm2bids / helper
+    helperfileList = os.listdir(helperPath)
+    if not helperfileList:
+        raise Exception('No data to process')
+
+    # Create list of acquisitions
+    acquisitionNames = list()
+    acquisitionNumbers = list()
+    iAcq = 0
+
+    # Create list containing all files
+    for iFile in range(len(helperfileList)):
+        # If it's a json file
+        name, ext = os.path.splitext(helperfileList[iFile])
+        if ext == '.json':
+            # Check for both.gz and .nii
+            booleanList = (np.in1d(helperfileList, name + '.nii') + np.in1d(helperfileList, name + '.nii.gz')).tolist()
```

Comment:
I'm not sure how your testing this, but a good way would to use isin instead of in2d that keeps the same shape of matrice and keeps your code coherent in your shapes to avoid indexes mistakes